### PR TITLE
Suffix and Prefix all keys

### PR DESCRIPTION
**Pull request by dav-m85**
Gives the ability to suffix and prefix all keys, like in:
```php
$statsdService->increment('bar');
# ... send bar:1|c

$statsdService->setSuffix('foo');
$statsdService->increment('bar');
# ... send foo.bar:1|c
```

### CODE REVIEW

File: tests/Liuggio/StatsdClient/StatsdDataFactoryTest.php [hunk 1]
```diff
@@ -94,4 +94,53 @@ public function testCreateStatsdDataUpdateCount()
         $this->assertEquals($stringValue, $obj->getValue());
         $this->assertEquals('c', $obj->getMetric());
     }
+
+
+    /**
+     * Prefix and suffix
+     */
+    public function testKeyMetricWithPrefixAndSuffix()
```
Comment (dav-m85) [inline tests/Liuggio/StatsdClient/StatsdDataFactoryTest.php]: This test is redundant. I shall remove it.

File: src/Liuggio/StatsdClient/Factory/StatsdDataFactory.php [hunk 2]
```diff
@@ -7,6 +7,13 @@
 class StatsdDataFactory implements StatsdDataFactoryInterface
 {
     /**
+     * @var string
+     */
+    protected $prefix = null;
```
Comment (dav-m85) [inline src/Liuggio/StatsdClient/Factory/StatsdDataFactory.php]: These properties shall be private.

File: src/Liuggio/StatsdClient/Factory/StatsdDataFactory.php [hunk 3]
```diff
@@ -127,4 +134,37 @@ public function getEntityClass()
...
+     * @return string
+     */
+    private function getKeyMetric($key)
+    {
+        if ($this->suffix !== null) $key = sprintf('%s.%s', $key, $this->suffix);
+        if ($this->prefix !== null) $key = sprintf('%s.%s', $this->prefix, $key);
+
+        return $key;
+    }
+}
```
Comment (dav-m85) [inline src/Liuggio/StatsdClient/Factory/StatsdDataFactory.php]: EOF

File: src/Liuggio/StatsdClient/Factory/StatsdDataFactoryInterface.php [hunk 4]
```diff
@@ -96,4 +96,24 @@ function updateCount($key, $delta);
...
+    /**
+     * Set an prefix key
+     *
+     * @abstract
+     *
+     * @param $prefix
+     * @return mixed
+     */
+    function setPrefix($prefix);
+}
```
Comment (dav-m85) [inline src/Liuggio/StatsdClient/Factory/StatsdDataFactoryInterface.php]: EOF

File: tests/Liuggio/StatsdClient/Service/StatsdServiceTest.php [hunk 5]
```diff
@@ -77,4 +81,4 @@ public function testSampling()
 
         $this->assertSame(0.1, $data->getSampleRate());
     }
-}
+}
```
Comment (dav-m85) [inline tests/Liuggio/StatsdClient/Service/StatsdServiceTest.php]: EOF :°O

**Comment by liuggio** [comment]
seems very good

**Pull request merged**